eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -481,11 +527,11 @@ object Router {
     }
 
     /** This method retrieves the channel update that we used when we built the route. */
-    def getChannelUpdateForNode(nodeId: PublicKey): Option[ChannelUpdate] = hops.find(_.nodeId == nodeId).map(_.lastUpdate)
+    def getChannelUpdateForNode(nodeId: PublicKey): Option[ChannelUpdate] = hops.find(_.nodeId == nodeId).map(_.params).collect { case s: ChannelRelayParams.FromAnnouncement => s.channelUpdate }
```

Comment:
The only place where we use this function is here: https://github.com/ACINQ/eclair/blob/ee74f10a8d956196dcc62ad297e83769aa7a2a82/eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala#L245

Your change shows that this was incorrect, we also need to handle the case where we receive a channel update for a channel that was in the invoice routing hints, don't we?

IMO we should remove this method, and instead more explicitly handle all cases in `PaymentLifecycle.scala`. WDYT?

Reply:
I agree and kind of let this for later. We should be able to rewrite `PaymentLifecycle.handleUpdate` much nicer now that we have streamlined channel sources, but I didn't want to dig an even deeper rabbit hole than the one I was in 😆 

Let's add a TODO and fix later?

Reply:
Right, that works for me :+1:

Reply:
Done with 6fe269081.